gym/vector/async_vector_env.py

```diff
@@ -0,0 +1,594 @@
+
+    episodic : bool (default: `False`)
+        If `True`, then the environments run for a single episode (until
+        `done=True`), and subsequent calls to `step` have an unexpected
+        behaviour. If `False`, then the environments call `reset` at the end of
+        each episode.
+
+    context : str, optional
+        Context for multiprocessing. If `None`, then the default context is used.
+        Only available in Python 3.
+    """
+    def __init__(self, env_fns, observation_space=None, action_space=None,
+                 shared_memory=True, copy=True, max_retries=0,
+                 episodic=False, context=None):
+        try:
+            self.ctx = mp.get_context(context)
+        except AttributeError:
+            logger.warn('Context switching for `multiprocessing` is not '
+                'available in Python 2. Using the default context.')
+            self.ctx = mp
+        self.env_fns = env_fns
+        self.shared_memory = shared_memory
+        self.copy = copy
+        self.max_retries = max_retries
+        self.episodic = episodic
+        self._num_retries = 0
+
+        if (observation_space is None) or (action_space is None):
+            dummy_env = env_fns[0]()
+            observation_space = observation_space or dummy_env.observation_space
+            action_space = action_space or dummy_env.action_space
+            dummy_env.close()
+            del dummy_env
+        super(AsyncVectorEnv, self).__init__(num_envs=len(env_fns),
+            observation_space=observation_space, action_space=action_space)
+
+        if self.shared_memory:
+            self._obs_buffer = create_shared_memory(
+                self.single_observation_space, n=self.num_envs)
+            self.observations = read_from_shared_memory(self._obs_buffer,
+                self.single_observation_space, n=self.num_envs)
+        else:
+            self._obs_buffer = None
+            self.observations = create_empty_array(
+                self.single_observation_space, n=self.num_envs, fn=np.zeros)
+
+        self.parent_pipes, self.processes = [], []
+        self.error_queue = self.ctx.Queue()
+        with clear_mpi_env_vars():
+            for index in range(self.num_envs):
+                process, parent_pipe = self._start_process(index)
+                self.parent_pipes.append(parent_pipe)
+                self.processes.append(process)
+
+        self._state = AsyncState.DEFAULT
+        self._check_observation_spaces()
+
+    def seed(self, seeds=None):
+        """
+        Parameters
+        ----------
+        seeds : list of int, or int, optional
+            Random seed for each individual environment. If `seeds` is a list of
+            length `num_envs`, then the items of the list are chosen as random
+            seeds. If `seeds` is an int, then each environment uses the random
+            seed `seeds + n`, where `n` is the index of the environment (between
+            `0` and `num_envs - 1`).
+        """
```

Comment:
@araffin The `seed` and `set_attr` methods can take lists of seeds/parameters to set different values for each individual environment. However this still requires the list to have the same length as the number of environments.